lib/jar.ts

```diff
@@ -138,11 +141,13 @@ async function getMavenPackageInfo(sha1: string): Promise<MavenPackageInfo> {
             ),
           );
         }
-
         if (res.response.docs.length > 1) {
-          debug('Got multiple results for sha1, only returning first one');
+          const sha1Target = path.parse(targetPath).base;
+          debug('Got multiple results for sha1, looking for', sha1Target);
+          res.response.docs.find((doc) =>
+            sha1Target.includes(doc.g) ? resolve(doc) : 0,
+          );
```

Comment:
```suggestion
          const foundPackage = res.response.docs.find(({g}) => sha1Target.includes(g));
          res.response.docs = [foundPackage] || res.response.docs;
```

Reply:
> 
res.response.docs = (foundPackage && [foundPackage]) || res.response.docs;

Reply:
`[foundPackage || res.response.docs[0]]`

Reply:
ok this one is sweetie